runtime/reference/security_model.md

```diff
@@ -0,0 +1,184 @@
+  explicitly to an escalation.
+- **The initial static module graph can import local files without
+  restrictions**: All files that are imported in the initial static module graph
+  can be imported without restrictions, so even if a explicit read permission is
+  not granted for that file. This does not apply to any dynamic module imports.
+
+These key principles are designed to provide an environment where a user can
+execute code with minimal risk of harm to the host machine or network. The
+security model is designed to be simple to understand and to provide a clear
+separation of concerns between the runtime and the code executing within it. The
+security model is enforced by the Deno runtime, and is not dependent on the
+underlying operating system.
+
+## Permissions
+
+By default, access to most system I/O is denied. There are some I/O operations
+that are allowed in a limited capacity, even by default. These are described
+below.
+
+To enable these operations, the user must explicitly grant permission to the
+Deno runtime. This is done by passing the `--allow-read`, `--allow-write`,
+`--allow-net`, `--allow-env`, and `--allow-run` flags to the `deno` command.
+
+During execution of a script, a user can also explicitly grant permission to
+specific files, directories, network addresses, environment variables, and
+subprocesses when prompted by the runtime. Prompts are not shown if
+stdout/stderr are not a TTY, or when the `--no-prompt` flag is passed to the
+`deno` command.
+
+### File system access
+
+By default, executing code can not read or write arbitrary files on the file
+system. This includes listing the contents of directories, checking for the
+existence of a given file, and opening or connecting to Unix sockets.
+
+Some APIs in Deno are implemented using file system operations under the hood,
+even though they do not provide direct read/write access to specific files. Some
+examples of these APIs are:
+
+- `localStorage`
+- Deno KV
+- `caches`
+- `Blob`
+
+Because these APIs are implemented using file system operations, users can use
+them to consume file system resources like storage space, even if they do not
+have direct access to the file system.
+
+During module loading, Deno can load files from disk. This sometimes requires
+explicit permissions, and sometimes is allowed by default:
+
+- All files that are imported from the entrypoint module in a way that they can
+  be statically analyzed are allowed to be read by default. This includes static
+  `import` statements and dynamic `import()` calls where the argument is a
+  string literal that points to a specific file or a directory of files. The
+  full list of files that are in this list can be printed using
+  `deno info <entrypoint>`.
+- Files that are dynamically imported in a way that can not be statically
+  analyzed require runtime read permissions.
+- Files inside of a `node_modules/` directory are allowed to be read by default.
```

Comment:
But not if used with `require()`, let's mention this here

Reply:
wdym?

Reply:
If you have `require()` calls in your code you will still need to add `--allow-read` flag

Reply:
Yeah, but not to read the `node_modules/` folder. You need it only to read the `package.json` in CWD. You can do this completely without any permissions for example:

```js
import { createRequire } from "node:module";

// import require from "c";
const require = createRequire(
  new URL("./node_modules/chalk", import.meta.url).href,
);

const chalk = require("chalk");
console.log(chalk.blue("Hello world!"));
```